packages/grafana-data/src/field/fieldOverrides.ts

```diff
@@ -119,7 +119,7 @@ export function applyFieldOverrides(options: ApplyFieldOverrideOptions): DataFra
         displayName,
       };
 
-      const config: FieldConfig = { ...field.config };
+      const config: FieldConfig = { ...cloneDeep(field.config) };
```

Comment:
@ifrost why are we doing a deep clone here?

Reply:
I've added it to fix switching visibility when toggling Logs legend in Explore.

For Logs `field.config` ([with specific colours and styles for logs](https://github.com/grafana/grafana/blob/main/public/app/core/logs_model.ts#L145)) is created only after the query is run, and clicking on the legend doesn't go through the reducer to create new field config, but only modifies the state of Logs component (and changes visibility of rows as well): https://github.com/grafana/grafana/blob/main/public/app/features/explore/Logs.tsx#L168

Without `cloneDeep` field config overrides are applied to the original `field.config` object that was created after the query was run and it's being reused.  It means that after making a series invisible for logs by clicking on the legend it stays invisible.

Does `deepClone` cause any problems? I thought adding it was something that would make sense anyway. I can take another look and maybe refactor it to a redux action that would change the field config on each legeng click and remove cloneDeep? During migration to GraphNG I wanted to re-use existing logic to minimise number of changes.